magefiles/magefile.go

```diff
@@ -369,6 +369,16 @@ func PublishImages() {
 	must.RunV("docker", "push", imgPermalink)
 }
 
+func PublishMultiArchImages() {
+	meta := releases.LoadMetadata()
+	img := Env.ManagerImagePrefix + meta.Version
+	imgPermalink := Env.ManagerImagePrefix + meta.Permalink
+
+	log.Printf("Multi-arch build and push of %s and %s\n", img, imgPermalink)
+	must.RunV("docker", "buildx", "create", "--use")
+	must.RunV("docker", "buildx", "bake", "-f", "docker-bake.json", "--push", "--set", "porter.tags="+img, "--set", "porter.tags="+imgPermalink, "porter")
```

Comment:
What do the porter.tags do? Just wondering who uses the tags and if they get published to the registry or baked into the image?

Reply:
Basically it's the equivalent of the `-t` arg to set tags for a build.  This is the beginnings of the value-add of the bulidx bake workflow.  A baker file (docker-bake.json) can contain the definition of docker builds of multiple patterns.  In this case the only target in the file is "porter" which is also the default.  (perhaps it should be changed to `porterop`? )  The [buildx baker](https://docs.docker.com/engine/reference/commandline/buildx_bake/)  pattern has the advantage of managing parallel builds as well.

The baker file can have targets with pre-defined tags, build context, etc.. Anything that would normally be part of a docker build.  The `--set` command is a general purpose arg for CRUD on the baker file at runtime.  Sort of like helm.

In this case there's no tags in the baker file for the `porter` target.  They are being set a runtime.  I see this as a starting place to have a single definition for build and publish patterns that can replace more of the docker build commands over time. I had an iteration of this work using it in more areas, but it would require finer grain mage targets to know when a image build is needed for a local kind load for test runs.  Multi-arch builds cannot load an image into the hosts cache for use by kind.   I wanted to get a final multi-arch publish in as an initial step. 

If this path looks desirable we can look at adopting it in the porter project as well.

Reply:
Neat! Yeah I'd be down for using this elsewhere. 👍